### ISSUE TITLE
[Regression] Carthage fails to build framework project from workspace, log is empty

### ISSUE THREAD

**Issue by jessesquires** [Carthage/Carthage#589]
#### Setup
````
Carthage 0.7.5
Xcode 7.0 (A121I) beta 2
````

#### Project
[JSQCoreDataKit](https://github.com/jessesquires/JSQCoreDataKit)
````
├── CONTRIBUTING.md
├── Example
│   ├── Example
│   ├── Example.xcodeproj
│   ├── ExampleModel
│   └── ExampleTests
├── JSQCoreDataKit
│   ├── JSQCoreDataKit
│   ├── JSQCoreDataKit.xcodeproj
│   ├── JSQCoreDataKitTests

├── JSQCoreDataKit.podspec
├── JSQCoreDataKit.xcworkspace
├── LICENSE
└── README.md
````

`JSQCoreDataKit.xcodeproj` is a Cocoa touch framework that builds/test independently. ` Example.xcodeproj` is an example app. Both projects are part of `JSQCoreDataKit.xcworkspace`.

#### Error
````
$ git clone https://github.com/jessesquires/JSQCoreDataKit.git
$ cd JSQCoreDataKit/
$ carthage build --no-skip-current
*** xcodebuild output can be found in /var/folders/tx/s__fjnns3yd_23813_gwk7sh0000gn/T/carthage-xcodebuild.b0hdgj.log
````
Carthage is no longer finding the scheme for `JSQCoreDataKit`. The log is empty. **This succeeded in previous version of Carthage**, I think `0.6.5`.

Interestingly, if you `cd JSQCoreDataKit/JSQCoreDataKit/` (the root directory of `JSQCoreDataKit.xcodeproj`) and run `carthage build --no-skip-current`, then it succeeds.

**Comment by jessesquires** [comment]
Confirmed: last working version is `0.7.3`

**Comment by esttorhe** [comment]
I can confirm; same thing is happening to me when trying to build `Moya`

Weird thing is it was working and then all of a sudden it just stopped building and now the log is empty.

**Comment by ikesyo** [comment]
@jessesquires This seems to be the same as #532, and PR #570 might be a help for that directory structure type.

**Comment by IamAlchemist** [comment]
ran into the same problem with carthage 0.7.5

**Comment by IamAlchemist** [comment]
similar happens to [this project](https://github.com/IamAlchemist/ScrollTabIndicator) in github.
i tried to revert carthage to 0.7.3, it still failed. and nothing in the log

**Comment by IamAlchemist** [comment]
I found a workaround, I recreate the workspace and add the projects again, then I run 'carthage build', it success with output in log

**Comment by ikesyo** [comment]
[0.8](https://github.com/Carthage/Carthage/releases/tag/0.8) is released, it would fix the issue (#570 is merged in it).

**Comment by jessesquires** [comment]
Confirmed that issue is fixed in `0.8` :tada:
